### ISSUE TITLE
The ScheduledJobsClient class improperly sets maxExecutionTimeInSeconds due to a bug in the ScheduledJobParserTest class

### ISSUE THREAD

**Issue by alanblythe** [Azure/azure-iot-sdk-java#1591]
# Context

- **OS and version used:** Ubuntu 20.04.3 LTS 
- **Java runtime used:** 1.8.0_342-8u342-b07-0ubuntu1~20.04-b07
- **SDK version used:** from latest source, Azure IoT SDK for Java Release 2022-7-25?

## Description of the issue
The ScheduledJobsClient class improperly sets maxExecutionTimeInSeconds due to a bug in the ScheduledJobParserTest class.
/service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/jobs/ScheduledJobsClient.java
/service/iot-service-client/src/test/java/com/microsoft/azure/sdk/iot/service/jobs/scheduled/ScheduledJobParserTest.java

This results in the maxExecutionTimeInSeconds being ignored as it's being sent as maxExecutionTimeSeconds
The REST API that's called is PUT /jobs/v2/{id}
https://docs.microsoft.com/en-us/rest/api/iothub/service/jobs/create-scheduled-job

## Code sample exhibiting the issue
I created a fork of the repo, modified the sample app to exhibit the problem. If you run the fork you will also have the code fix as well though.
https://github.com/alanblythe/azure-iot-sdk-java/tree/devcontainer-changes

## Console log of the issue
No Logs

## Fix  
The following commit on the fork shows how to reproduce, the sample app exhibits the problem
https://github.com/alanblythe/azure-iot-sdk-java/commit/aa0fbdccdc093d4f06866cbe206c6b2d0e849e3f
using the following code will cause improper serialization of maxExecutionTimeInSeconds  

```
        DirectMethodsJobOptions options =
            DirectMethodsJobOptions.builder()
                .payload(payload)
                .methodConnectTimeoutSeconds(connectTimeout)
                .methodResponseTimeoutSeconds(responseTimeout)
                .maxExecutionTimeSeconds(maxExecutionTimeInSeconds)
                .build();

        ScheduledJob jobMethod =
            jobClient.scheduleDirectMethod(
                jobIdMethod,
                queryCondition,
                methodName,
                startTimeUtc,
                options);

```

I have a fix on my fork that I can cherry pick/commit to a branch on the main repo
I erred on the side of caution and avoided breaking changes.
The improper name will still exist in certain places such as in DirectMethodsJobOptions and any class that uses it. 
azure-iot-sdk-java/service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/jobs/DirectMethodsJobOptions.java
A full list here: https://github.com/Azure/azure-iot-sdk-java/search?q=maxexecutiontimeseconds

**Comment by alanblythe** [comment]
moved the modified sample app from the fork/main branch to the devcontainer-changes branch. I updated the issue description for this updated link.
https://github.com/alanblythe/azure-iot-sdk-java/tree/devcontainer-changes

**Comment by ngastelum-ms** [comment]
Fix included in version 2.1.0
